SSO Service Name Migration (#594)
BCDevOps/developer-experience: docs/kcMigration/sso-migration-notification.md

Once the service name enablment is complete, SSO service will return back to normal. What you should be expecting is that KeyCloak would then be accessible via both service names, `sso-*.pathfinder.gov.bc.ca` and `*.oidc.gov.bc.ca`.

### Do I need to do anything?

You should check on all existing usages of SSO KeyCloak after the new name has been added to confirm your use of the sso-dev.pathfinder.gov.bc.ca service name is not impacted. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.

Once the change has been completed in an environment, you can begin to update the SSO service name for all application integrations and test out the authentication flow with the new enterprise service name. (If you have BCSC integration with your realm, stay tuned for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)

> Here are the recommended ways to start with:

Reviewer: ```suggestion
### Here are the recommended ways to start with:
```

### What is happening?

> Hi team, the KeyCloak SSO service is finally removing its *Pathfinder* sticker and getting a new name.  This change does not affect the service offering but does change how you will access it.  The name is changing from sso.pathfinder.gov.bc.ca to oidc.gov.bc.ca.  For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).

Reviewer 2: ```suggestion
Hi team, the KeyCloak SSO service is finally starting the process of removing its *Pathfinder* sticker and getting a new name.  This change does not affect the service offering but will eventually change how you will access it.  The name is changing from sso.pathfinder.gov.bc.ca to oidc.gov.bc.ca.  For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).
```

- Step 3 - Product teams update application's SSO configuration to the new service and test out SSO integrations
- Step 4 - Platform Services team work with product teams to troubleshot any issue found
- Step 5 - Shutdown the pathfinder service name

#### Dev SSO New Service Name Enable on `Sep 24th`
As no production applications will be affected, new service name update will be scheduled during work hours. The expected SSO service disruption will last for 1.5 to 2 hours.

1 week of time are set aside for Dev environment testing before the Test environment service name change.

#### Test SSO New Service Name Enable on `Oct 1st`
As no production application will be affected, new service name update will be scheduled during work hours. The expected SSO service disruption will last for 1.5 to 2 hours.

1 week of time are set aside for Test environment testing before the Production environment service name change.

#### Production SSO New Service Name Enable on `Oct 8th`
Current SSO service disruption for the new production service name is scheduled for 5:30PM - 7:30PM (2 hours), on **October 8th, 2020**.

### Will there be an impact on the Platform apps?

> Yes there will be service disruption when we enable the new SSO service name, but the effects are expected to be temporary and will last for about 10mins for each realm. This is caused by KeyCloak internal settings update from Platform Services team, if you are interested in more details, refer to [doc](./KeycloakMigration.md).

Reviewer 2: is this in addition to the 1.5 - 2 hour service disruptions mentioned above?

Author: no it's included. the 1.5 to 2 hours includes the 10 mins disruption for each realm.

Also note that production SSO service disruption will be scheduled outside of business hours, after 5:00PM.

Once the service name enablment is complete, SSO service will return back to normal. What you should be expecting is that KeyCloak would then be accessible via both service names, `sso-*.pathfinder.gov.bc.ca` and `*.oidc.gov.bc.ca`.

### Do I need to do anything?

> Yes, you will need to participate in both `Step 2 & 3` after new service name is enabled.

Reviewer 2: Cleaning up after removing step 3
```suggestion
```

> 

Reviewer 2: ```suggestion
```

> For Step 2, you should check on all existing usages of SSO KeyCloak immediately. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.

Reviewer 2: ```suggestion
You should check on all existing usages of SSO KeyCloak after the new name has been added to confirm your use of the sso-dev.pathfinder.gov.bc.ca service name is not impacted. There will be __one week__ to report any bugs or integration issues, before we move onto the next environment. Platform Service team will able available to help troubleshoot issues relevant to the service name change when any SSO admin action is needed.
```

> For Step 3, you should start to update the SSO service name for all application integrations and test out the authentication flow as soon as possible. (If you have BCSC integration with your realm, stay tune for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)

Reviewer 2: ```suggestion
Once the change has been completed in an environment, you can begin to update the SSO service name for all application integrations and test out the authentication flow with the new enterprise service name. (If you have BCSC integration with your realm, stay tuned for more details on how to proceed with the update. We will be working with some polit teams and BCSC team once Dev SSO has the new service name in place.)
```
